build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -841,12 +851,13 @@ message ActionResult {
   // The exit code of the command.
   int32 exit_code = 4;
 
-  // The standard output buffer of the action. The server will determine, based
-  // on the size of the buffer, whether to return it in raw form or to return
-  // a digest in `stdout_digest` that points to the buffer. If neither is set,
-  // then the buffer is empty. The client SHOULD NOT assume it will get one of
-  // the raw buffer or a digest on any given request and should be prepared to
-  // handle either.
+  // The standard output buffer of the action. The server MUST NOT inline
+  // stdout unless requested by the client in the
+  // [GetActionResultRequest][build.bazel.remote.execution.v2.GetActionResultRequest]
+  // message and instead return a digest in `stdout_digest` that points to
+  // the buffer. The client SHOULD NOT assume it will get one of the raw
```

Comment:
Somewhat orthogonal to your proposal, but I think we err'd in saying that stdout_raw may be set and stdout_digest then left empty. I'd prefer the semantics be that stdout_digest is always set, and stdout_raw may *also* be set if inlining is requested. Otherwise if a client both wants to display the stdout and save a reference to it for later, the client needs to *upload* it to the CAS for persistence themselves, which seems backwards.

If we're changing the semantics of inlining anyways, and no implementation we know of relies on the semantics of today, could we change this?

Reply:
I agree that it would be better to report the bytes and the digest.

Reply:
Great idea. Done.